implementations/rust/ockam/ockam_api/src/cli_state.rs

```diff
@@ -171,6 +171,19 @@ impl VaultsState {
         std::os::unix::fs::symlink(original, link)?;
         self.get(name)
     }
+
+    pub fn list(&self) -> Result<Vec<VaultState>> {
+        let mut vaults: Vec<VaultState> = vec![];
+        for entry in std::fs::read_dir(&self.dir)? {
+            let entry = entry?;
+            if entry.path().extension() == Some(OsStr::from_bytes(b"json")) {
+                let file_name = entry.path();
+                let vault_name = file_name.file_stem().unwrap().to_str().unwrap();
```

Comment:
There is a utils function called `file_stem(&entry_path())` that does just that.

Reply:
`list()` was already added, including this suggestion.